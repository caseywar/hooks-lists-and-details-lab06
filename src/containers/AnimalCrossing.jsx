import React, { useState, useEffect } from 'react';
import VillagerList from '../components/villagers/VillagerList'
import { fetchVillagers } from '../services/AnimalCrossingApi'

const AnimalCrossing = () => {
    const [loading, setLoading] = useState(true);
    const [villagers, setCillagers] = useState([]);

    useEffect(() => {
        fetchVillagers()

        .then(setCillagers)
        .finally(() => setLoading(false));
    }, []);

    if (loading) return <h1>Loading...</h1>;

    return (
        <VillagerList villagers={villagers} />
    );
};

export default AnimalCrossing;