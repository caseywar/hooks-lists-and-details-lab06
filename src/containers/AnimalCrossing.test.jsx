import React from 'react';
import { render, screen } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import mockVillagerData from '../../fixtures/villagers.json';
import AnimalCrossing from './AnimalCrossing'
import { MemoryRouter } from 'react-router';


const server = setupServer(
    rest.get('https://ac-vill.herokuapp.com/villagers/', (req, res,ctx) => {
        return res(ctx.json(mockVillagerData));
    })
);

describe('Animal Crossing container', () => {
    beforeAll(() => server.listen());
    afterAll(() => server.close());

    it('displays the loading screen', async () => {
        render(
            <MemoryRouter>
            <AnimalCrossing />
            </MemoryRouter>);

        screen.getByText('Loading...');
    });

    it('desiplays a list of villagers', async () => {
        render(
            <MemoryRouter>
            <AnimalCrossing />
            </MemoryRouter>);

        const ul = await screen.findAllByRole('list', { name: 'villagers' });
        expect(ul).toMatchSnapshot();
    })
})