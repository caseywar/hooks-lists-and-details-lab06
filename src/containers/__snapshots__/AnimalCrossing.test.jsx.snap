// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animal Crossing container desiplays a list of villagers 1`] = `
Array [
  <ul
    aria-label="villagers"
  >
    <li>
      <div>
        <img
          alt="Admiral"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/ed/Admiral_NH.png/revision/latest?cb=20200802081138"
        />
        <p>
          Admiral
        </p>
        <p>
          イッテツ Ittetsu
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Agent S"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/a/a7/Agent_S_NH.png/revision/latest?cb=20200719092526"
        />
        <p>
          Agent S
        </p>
        <p>
          2ごう Ni Gō
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Agnes"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/4/42/Agnes_NH.png/revision/latest?cb=20200804230432"
        />
        <p>
          Agnes
        </p>
        <p>
          アグネス Agunesu
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Al"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/c4/Al_NH.png/revision/latest?cb=20200802150354"
        />
        <p>
          Al
        </p>
        <p>
          たもつ Tamotsu
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Alfonso"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/e4/Alfonso_NH.png/revision/latest?cb=20200817164717"
        />
        <p>
          Alfonso
        </p>
        <p>
          アルベルト Aruberuto
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Alice"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/e1/Alice_NH.png/revision/latest?cb=20200803181400"
        />
        <p>
          Alice
        </p>
        <p>
          メルボルン Meruborun
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Ankha"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/56/Ankha_NH.png/revision/latest?cb=20200817172044"
        />
        <p>
          Ankha
        </p>
        <p>
          ナイル Nairu
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Angus"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/5f/Angus_NH.png/revision/latest?cb=20200802083758"
        />
        <p>
          Angus
        </p>
        <p>
          セルバンテス Serubantesu
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Alli"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/8/8e/Alli_NH.png/revision/latest?cb=20200802061731"
        />
        <p>
          Alli
        </p>
        <p>
          クロコ Kuroko
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Anicotti"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/8/80/Anicotti_NH.png/revision/latest?cb=20200802164003"
        />
        <p>
          Anicotti
        </p>
        <p>
          ラザニア Razania
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Amelia"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/a/ab/Amelia_NH.png/revision/latest?cb=20200817171545"
        />
        <p>
          Amelia
        </p>
        <p>
          アンデス Andesu
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Annalisa"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/f/f0/Annalisa_NH.png/revision/latest?cb=20200802063308"
        />
        <p>
          Annalisa
        </p>
        <p>
          みやび Miyabi
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Antonio"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d4/Antonio_NH.png/revision/latest?cb=20200802063352"
        />
        <p>
          Antonio
        </p>
        <p>
          マコト Makoto
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Annalise"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/c4/Annalise_NH.png/revision/latest?cb=20200802153943"
        />
        <p>
          Annalise
        </p>
        <p>
          シルブプレ Shirubupure
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Anabelle"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d8/Anabelle_NH.png/revision/latest?cb=20200802063200"
        />
        <p>
          Anabelle
        </p>
        <p>
          あるみ Arumi
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Apollo"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/2/28/Apollo_NH.png/revision/latest?cb=20200719092927"
        />
        <p>
          Apollo
        </p>
        <p>
          アポロ Aporo
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Anchovy"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/ea/Anchovy_NH.png/revision/latest?cb=20200802081231"
        />
        <p>
          Anchovy
        </p>
        <p>
          アンチョビ Anchobi
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Apple"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/57/Apple_NH.png/revision/latest?cb=20200802151508"
        />
        <p>
          Apple
        </p>
        <p>
          アップル Appuru
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Astrid"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d5/Astrid_NH.png/revision/latest?cb=20200802154954"
        />
        <p>
          Astrid
        </p>
        <p>
          キッズ Kizzu
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Audie"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/1/1b/Audie_NH.png/revision/latest?cb=20200829012735"
        />
        <p>
          Audie
        </p>
        <p>
          モニカ Monika
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Aurora"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/59/Aurora_NH.png/revision/latest?cb=20200202215217"
        />
        <p>
          Aurora
        </p>
        <p>
          オーロラ Ōrora
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Avery"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/dd/Avery_NH.png/revision/latest?cb=20200608222354"
        />
        <p>
          Avery
        </p>
        <p>
          クスケチャ Kusukecha
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Ava"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/cc/Ava_NH.png/revision/latest?cb=20200609055741"
        />
        <p>
          Ava
        </p>
        <p>
          ドミグラ Domigura
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Barold"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/9/9d/Barold_NH.png/revision/latest?cb=20200802141140"
        />
        <p>
          Barold
        </p>
        <p>
          ニッシー Nisshī
        </p>
      </div>
    </li>
    <li>
      <div>
        <img
          alt="Bangle"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/6/68/Bangle_NH.png/revision/latest?cb=20200803120044"
        />
        <p>
          Bangle
        </p>
        <p>
          ルーズ Rūzu
        </p>
      </div>
    </li>
  </ul>,
]
`;
